lib/fizzy/execute.cpp

```diff
@@ -323,15 +323,20 @@ const FuncType& function_type(const Instance& instance, FuncIdx idx)
 }
 
 bool invoke_function(const FuncType& func_type, uint32_t func_idx, Instance& instance,
-    Stack<uint64_t>& stack, int depth)
+    OperandStack& stack, int depth)
 {
     if (depth == CallStackLimit)
         return false;
 
     const auto num_args = func_type.inputs.size();
     assert(stack.size() >= num_args);
-    std::vector<uint64_t> call_args(stack.end() - static_cast<ptrdiff_t>(num_args), stack.end());
-    stack.resize(stack.size() - num_args);
+    std::vector<uint64_t> call_args;
+    if (num_args != 0)
+    {
+        call_args.reserve(num_args);
+        std::copy_n(&stack[num_args - 1], num_args, std::back_inserter(call_args));
+    }
+    stack.shrink(stack.size() - num_args);
```

Comment:
if you stay with `if (num_args != 0)`, this line can be inside

Reply:
Can it?

Reply:
Yes. I unsuccessfully pushed this change last week.